README.md

```diff
@@ -66,6 +66,14 @@ Following are the implementations managed by third-parties adopting the standard
   * Machine API Operator, https://github.com/openshift/machine-api-operator/tree/master
   * Machine-controller-manager, https://github.com/gardener/machine-controller-manager/tree/cluster-api
 
+## Versioning, Maintenance, and Compatibility
+
+- We follow [Semantic Versioning (semver)](https://semver.org/).
+- Releases/Milestones are set 6 weeks after a Kubernetes release.
```

Comment:
The way this is phrased makes me think we will be discussing milestone dates 6 weeks after each kubernetes release. I think if you change "are set" to just "are" that would make it match my understanding of what we said yesterday in the meeting.

Reply:
Has the time frame already been debated/discussed or is this a placeholder for now?

Reply:
It was discussed in the meeting this week, but I think it's fair to discuss it more here as well.

Reply:
Is the idea here that the 6 weeks will allow time to develop against the newly released Kubernetes version? In [talos](https://github.com/talos-systems/talos) we aim to develop the next version as the next version of Kubernetes is developed. That way we can land support only days after a Kubernetes release.

Reply:
Ideally, infrastructure and bootstrap providers could provide updated releases to add support for newer Kubernetes versions without requiring a new version of the `cluster-api` repository. I could imagine something like this:

cluster-api API version: v1alpha2
Kubernetes version: v1.16
talos provider API version: v1alpha2, semver v0.5.0

Now Kubernetes v1.17 comes out. The talos provider could release a new semver v0.5.1 that adds support for Kubernetes v1.17. The API version would remain at v1alpha2 for both talos and cluster-api. The only way this wouldn't work is if there are data model changes to the talos API types to be able to handle Kubernetes v1.17. Then you'd want a new API version...

6 weeks after Kubernetes v1.17 is released, a new cluster-api with v1alpha3 comes out, along with updated releases for the various infra and bootstrap providers.

Do you all think something like this would make sense?

Reply:
@ncdc I think this should be fine. The only thing I'm a bit confused about is how this works if in our Talos provider we want k8s 1.17, but https://github.com/kubernetes-sigs/cluster-api/pull/967/files#r289097715 requires 1.16. For example we currently use k8s 1.14 in Talos, but we can't pull in any Talos code into our provider because the provider depends on 1.13: https://github.com/talos-systems/cluster-api-provider-talos/blob/master/Gopkg.lock#L871.

Reply:
Maybe we could chat about this on Slack or Zoom? I'm not familiar with Talos and how/where it uses Kubernetes.

Reply:
Works for me. I will find you in k8s slack and send you a zoom link.

Reply:
Fixed the wording a little, PTAL!